proposals/0000-a11y-roles-states-improvements.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Improvements to accessibilityRole and accessibilityStates
+author:
+- Marc Mulcahy
+date: 2018-10-31
+---
+
+## Summary
+
+Some brilliant work was begun to add cross platform support for accessibility roles and states to React Native. We propose to add some additional roles and states, as well as extend the implementation on Android and Fire OS.
+
+## Basic example
+
+With the current set of roles, it is not possible to describe a checkbox. These changes would enable this scenario, among other things:
+
+    <View
+      accessibilityRole="checkbox"
+      accessibilityStates={['unchecked']}>
```

Comment:
Would `'unchecked'` be drawn from an enumerable?

Reply:
I'd prefer not to go this route, as it makes implementation a lot more complex, in that each state has to be an object with variable type. Any reason to add such complexity?

Reply:
@marcmulcahy as long as the strings are arbitrary. If not, if field values are drawn from a specific set, then without an enum, the functionality is simply based on convention.

Reply:
@marcmulcahy I think I'm imposing a implementation detail on the spec. I retract these comments.